virtcontainers/qemu.go

```diff
@@ -1363,7 +1366,20 @@ func (q *qemu) hotplugAddMemory(memDev *memoryDevice) (int, error) {
 		}
 		memDev.slot = maxSlot + 1
 	}
-	err = q.qmpMonitorCh.qmp.ExecHotplugMemory(q.qmpMonitorCh.ctx, "memory-backend-ram", "mem"+strconv.Itoa(memDev.slot), "", memDev.sizeMB)
+	if q.qemuConfig.Knobs.HugePages {
+		// we are setting all the bits that govmm sets when hugepages are enabled.
+		// https://github.com/intel/govmm/blob/master/qemu/qemu.go#L1677
+		target = "/dev/hugepages"
```

Comment:
Should we just go away from explicit huge pages and use the memory-backend-file method long term.

Reply:
I think so. Even if this can be the most performant way to use virtio-fs, it can have expectation on the host that most people don't have, therefore I think only people who know what they're doing should enable it.

Reply:
Today from master and 1.7.x release, virtio-fs uses `/dev/shm` and is no longer using hugepages. this change is only making sure we are compatible with hugepages and memory hotplug.